### PR TITLE
Enable and fix Windows tests

### PR DESCRIPTION
**Pull request by macisamuele**
This PR does introduce multiple features:
 1) Moves testing from TravisCI to GitHub Actions
 2) Enable testing for all the platforms
 3) Fixes an unknown bug on the Windows Platform (only "real" library change)
 4) Enable automated release when a tag is added
 5) Bump all pre-commit hooks

**NOTE**: This PR could still be merged even if we're not going to have automated release enabled.

@sjaensch: I don't have access to the repository setting and in order to have automated release enabled we need to define a secrets in there

The guide for doing it is available in [here](https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#saving-credentials-on-github).

Github actions run: https://github.com/macisamuele/swagger-spec-compatibility/actions/runs/325781459

---

The TL;DR; is:
 * we need to create an API token from a pypi.org account that can release the package
 * store it into the github secret that has to be named `pypi_password`

The advantage of using the API token is that we can easily de-activate it if for some reason it get's exposed to the public and it is un-related to your normal passwords. Essentially it also gives you the flexibility to de-activate tokens for individual projects (which is something that I tend to do on the rust libs that I'mw working on).

Last note: github secrets are not visible (on the UI) once they are saved, but members of the org with enough permissions might be read the secret via APIs or by hardly debugging a run on the github worker. For this reason I would never-ever suggest to use your password but rather a created token.

### CODE REVIEW

File: .github/workflows/build.yml
Comment (cortinico) [inline .github/workflows/build.yml:3]:
```suggestion
on: [push, pull_request]
```

File: README.rst
Comment (cortinico) [inline README.rst:1]:
```suggestion
.. image:: https://github.com/Yelp/swagger-spec-compatibility/workflows/ci/badge.svg
```

File: .github/workflows/build.yml
Comment (cortinico) [inline .github/workflows/build.yml:4]: You need to remove the following two lines

Comment (macisamuele) [reply .github/workflows/build.yml:4]: lol ... I've just accepted your suggestions :( My bad on not checking if the other lines would be removed ;)

Comment (macisamuele) [reply .github/workflows/build.yml:4]: But is interesting how tests are essentially not run in here.
@cortinico do you have ideas on why this is happening (or better not happening)?

Comment (cortinico) [reply .github/workflows/build.yml]: > @cortinico do you have ideas on why this is happening (or better not happening)?

I think it's either:

1. The settings for this repo are not allowing Github Actions coming from PRs. If this is the case we should probably ask @sjaensch to look into it.
2. This is the first PR that is adding a Github Action. It will work once you merge this.

Comment (macisamuele) [reply .github/workflows/build.yml]: I'd vote for the first option ;)

File: .github/workflows/build.yml
Comment (cortinico) [inline .github/workflows/build.yml:20]:
```suggestion
    - name: Run docs tox job
```

**Review by cortinico — APPROVED** [review]
LGTM for the Github Actions part

**Review by cortinico — APPROVED** [review]

**Comment by macisamuele** [comment]
@analogue, @sjaensch could you please have a look to this PR?
Merging this would allow to have green tests on the repository.

**Comment by sjaensch** [comment]
I'm going to defer to @analogue as Core Services owns this library, and this PR both changes how tests are executed and introduces a way to do automated releases.

**Comment by macisamuele** [comment]
That works for me. About automated releases, if we find more involved to enable it on Github Actions we could still keep travis doing it.

cc @piax93: whom supported us to enable automated releases via travis (provided by [yelplabs](https://pypi.org/user/yelplabs/))

**Comment by piax93** [comment]
Hi, I do still have access to the `yelplabs` user and I seem to be able to create an API token for the project. I can work with @analogue to set that up when this gets approved.

**Comment by analogue** [comment]
@macisamuele Thanks for putting this together. I'm going to have to take some time to research the changes here and probably pull in release engineering for their wider thoughts wrt moving off TravisCI. Ideally, I'd like to keep things consistent across opensource projects and not having "one off" build workflows per git repo.

cc: @kkellyy

**Comment by macisamuele** [comment]
@analogue , @kkellyy , @chriskuehl: do we have any update on this?

**Comment by macisamuele** [comment]
@analogue do we have any update on this? As for now automated CI cannot successfully run on this project.

I see that https://github.com/Yelp/bravado-core/pull/380 enables github actions on bravado-core, so I do assume (and I might be wrong) that that there is an interest to move toward github actions

**Comment by analogue** [comment]
@macisamuele Yes, we'd like to migrate from Travis to GH Actions now :) I'll go ahead and merge this and see what else is necessary to get the automated release process to work.

**Comment by macisamuele** [comment]
@analogue  thanks a lot for merging this. i see now that pre-commit are failing. Going to fix them right away.

About the automated release it is documented in https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#saving-credentials-on-github.
The details to follow are
> The TL;DR; is:
> we need to create an API token from a pypi.org account that can release the package
> store it into the github secret that has to be named pypi_password

NOTE: publish.yaml workflow does already take care of the release, assuming that the secret would be `pypi_password` secret would be defined.

**Comment by analogue** [comment]
I don't have perms to manage this project on PyPI atm so expect some delay.

**Pull request merged**
